scanpy/neighbors/__init__.py

```diff
@@ -978,8 +981,13 @@ def compute_eigen(
             which = 'LM' if sort == 'decrease' else 'SM'
             # it pays off to increase the stability with a bit more precision
             matrix = matrix.astype(np.float64)
+
+            # Setting the random initial vector
+            random_state = check_random_state(random_state)
+            np.random.set_state(random_state.get_state())
```

Comment:
Is this line necessary for reproducibility? If not, I don't think we have to modify the global state.

Reply:
I've just checked, specifying the v0 is enough to get the same results every time. No need for setting global state. I will delete the line, thanks!